src/components/LinkableText.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {Text, Linking} from 'react-native';
+import {COLORS} from '../utils/theme';
+
+interface Props {
+  text: string;
+  style?: object;
+}
+
+export const LinkableText: React.FC<Props> = ({text, ...props}) => {
+  const handlePress = (url: string) => {
+    Linking.openURL(url);
+  };
+
+  // Regular expression to match URLs
+  const urlRegex = /(https?:\/\/[^\s]+)/g;
+
+  // Split the text by URLs
+  const parts = text.split(urlRegex);
+
+  return (
+    <Text style={props.style}>
+      {parts.map((part, index) =>
+        urlRegex.test(part) ? (
+          <Text
+            key={index}
+            style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
+            onPress={() => handlePress(part)}>
+            {part}
+          </Text>
+        ) : (
+          <Text key={index}>{part}</Text>
+        )
+      )}
+    </Text>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `LinkableText` component looks good overall. However, there's a potential issue with the regular expression test in line 24. The `test()` method advances the starting position of the next match after each successful match. This means that if you call `test()` multiple times on the same regex object, it may not start from the beginning of the string. To avoid this, you should create a new RegExp object for each test.

```diff
-        urlRegex.test(part) ? (
+        new RegExp(urlRegex).test(part) ? (
```

Also, consider adding error handling to the `handlePress` function. If the URL is malformed or if there's no registered application for the URL scheme, `Linking.openURL(url)` will throw an error.

```diff
  const handlePress = (url: string) => {
    Linking.openURL(url);
+    .catch((error) => console.error('An error occurred', error));
  };
```
Lastly, using the index as a key in React can cause issues if the order of items changes. It would be better to use a unique value related to the data, like the part itself.

```diff
-           <Text key={index}>{part}</Text>
+           <Text key={part}>{part}</Text
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->